src/OAuth2/Encryption/FirebaseJwt.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace OAuth2\Encryption;
+
+/**
+ * Bridge file to use the firebase/php-jwt package for JWT encoding and decoding.
+ * @author Francis Chuang <francis.chuang@gmail.com>
+ */
+class FirebaseJwt implements EncryptionInterface
+{
+    public function __construct()
+    {
+        if (!class_exists('\JWT')) {
+            throw new \ErrorException('firebase/php-jwt must be installed to use this feature. You can do this by running "composer require firebase/php-jwt"');
+        }
+    }
+
+    public function encode($payload, $key, $alg = 'HS256', $keyId = null)
+    {
+        return \JWT::encode($payload, $key, $alg, $keyId);
+    }
+
+    public function decode($jwt, $key = null, $allowedAlgorithms = null)
+    {
+        try {
+
+            //Maintain BC: Do not verify if no algorithms are passed in.

```

Comment:
If that's the case, then we will need to change the default value of `$allowedAlgorithms` to `array()` and break the interface. Another issue is due to the design of the original JWT library and the recent work to fix the security issue, the last parameter `$allowedAlgorithms` is conflated to provide a list of supported algorithms or to control whether the JWT should be verified or not. For `firebase/php-jwt`, the presence of the `$key` value controls whether the JWT should be verified or not.